### PR TITLE
fix: jurisdiction permission array approach

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #3639 & https://github.com/metrotranscom/doorway/issues/326

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR updates the ways listings approval permissions are determined (from env to db). Now, each jurisdiction has a listingsApprovalPermissions filed which stores an array. A null or empty array indicates the field is off and if users are listed, they are the approving roles in the listings approval flow. This allows for easier management of multiple jurisdictions having unique needs. The majority of the lines changes here, however, are related to backend testing and ensuring the functionality of the email services connected to this feature. 
 
## How Can This Be Tested/Reviewed?

Ensure that the previous functionality of the listings approval flow works and specifically, ensure that a jurisdictional admin can create a pending review application and can see different experiences based on the jurisdiction that they are assigned too. 

Note: there is one future consideration in that if a partner can create listings, we'd have to tie them to a jurisdiction in some way.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/test/listings/listings.e2e-spec.ts [hunk 1]
```diff
@@ -450,6 +472,175 @@ describe("Listings", () => {
...
+        .expect(200)
+
+      expect(listingPendingApprovalResponse.body.status).toBe(ListingStatus.pendingReview)
+      expect(mockRequestApproval).toBeCalledWith(
+        expect.objectContaining({
+          id: adminId,
+        }),
+        { id: listing.id, name: listing.name },
+        expect.arrayContaining(["admin@example.com", "mfauser@bloom.com"]),
+        "http://localhost:3001"
```
Comment (ludtkemorgan) [inline backend/core/test/listings/listings.e2e-spec.ts:512]: I'd recommend not hardcoding this to the url and rather have it be `process.env.PARTNERS_PORTAL_URL`. Same with the other tests that have this. That way these tests will pass no matter what the url is set to in the environment it is in.

**Review by ludtkemorgan — COMMENTED** [review]
Looks good. Just some minor comments

**Comment by ColinBuyck** [comment]
Updated @ludtkemorgan! Thanks for the feedback 🐧

File: sites/partners/src/components/listings/ListingFormActions.tsx [hunk 2]
```diff
@@ -234,9 +249,8 @@ const ListingFormActions = ({
             if (type === ListingFormActionsType.edit) {
               submitFormWithStatus(false, ListingStatus.active)
             } else {
-              // button only exists for listings approval so can call updateAndNotify directly
               try {
-                const result = await listingsService.updateAndNotify({
+                const result = await listingsService.update({
```
Comment (emilyjablonski) [inline sites/partners/src/components/listings/ListingFormActions.tsx:253]: I dig this shift

**Review by emilyjablonski — APPROVED** [review]

**Pull request merged**